### PR TITLE
Add Yamato configs

### PR DESCRIPTION
**Pull request by ashwinimurt**
Tested on commit before squash: https://yamato.prd.cds.internal.unity3d.com/jobs/81-mono/tree/unity-master-yamato/.yamato%252Fcollate_builds.yml/96720/job/pipeline

### CODE REVIEW

**Comment by ashwinimurt** [comment]
Kicked off Yamato builds on squashed commit. I will link the pipeline before merge.

**Comment by joshpeterson** [comment]
Can we add the Yamato builds as checks for Github PRs?

File: .yamato/build_linux_x64.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+sudo apt-get install -y schroot
+sudo apt-get install -y binutils debootstrap
+git submodule update --init --recursive
+git submodule update --init --recursive
```
Comment (joncham) [inline .yamato/build_linux_x64.sh]: This is here twice.

File: .yamato/build_linux_x86.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+sudo apt-get install -y schroot
+sudo apt-get install -y binutils debootstrap
+sudo apt-get install -y m4
+sudo apt-get install -y libc6-i386
+sudo apt-get install -y libc6-dev-i386
+sudo apt-get install -y libncurses5-i386
+git submodule update --init --recursive
+git submodule update --init --recursive
```
Comment (joncham) [inline .yamato/build_linux_x86.sh]: Same duplication.

File: .yamato/build_win.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+agent:
+  type: Unity::VM
+  image: platform-foundation/windows-mono-bokken:latest
+  flavor: b1.xlarge
+
+interpreter: powershell
+
+commands:
+  - git submodule update --init --recursive
```
Comment (joncham) [inline .yamato/build_win.yml]: Might be nice to have windows like other platforms where all logic is in a script, and the yml calls a single command. I think this also helps capture all output.

File: .yamato/build_linux_x64.sh (hunk 1 above)
Comment (ashwinimurt) [reply .yamato/build_linux_x64.sh]: Its failing sometimes due to network issues and retry also fails. So manually trying twice. Only on linux.

Comment (joncham) [reply .yamato/build_linux_x64.sh]: Then minor, but can we use loop and check exit code a few times?

File: .yamato/build_win.yml (hunk 3 above)
Comment (ashwinimurt) [reply .yamato/build_win.yml]: Updated

**Pull request merged**